threading.h

```diff
@@ -92,13 +95,15 @@ namespace tgvoip{
 		static void* ActualEntryPoint(void* arg){
 			Thread* self=reinterpret_cast<Thread*>(arg);
 			if(self->name){
-#if !defined(__APPLE__) && !defined(__gnu_hurd__)
+#if defined(__linux__) || defined(__FreeBSD__)
```

Comment:
I don't like the idea of using positive logic here

Reply:
Pardon tautology, can you please be more positive in your reviews? "_I don't like it_" doesn't explain much. Let me explain what I dislike in the negative logic here. Here is original code:

```
#if !defined(__APPLE__) && !defined(__gnu_hurd__)
				pthread_setname_np(self->thread, self->name);
#elif !defined(__gnu_hurd__)
				pthread_setname_np(self->name);
				if(self->maxPriority){
					DarwinSpecific::SetCurrentThreadPriority(DarwinSpecific::THREAD_PRIO_USER_INTERACTIVE);
				}
#endif
```
What we read here is that "if it is not GNU Hurd, then it is Darwin", which isn't a valid logical conclusion. My version is a straightforward switch/case statement.

Again, if you insist, this patch can be removed from the pull request.

Reply:
> What we read here is that "if it is not GNU Hurd, then it is Darwin", which isn't a valid logical conclusion.

That way the code can be still used by other UNIX/POSIX systems like other BSDs and Haiku. If you will make the logic positive, this will be broken for them.

Reply:
> Pardon tautology, can you please be more positive in your reviews?

I don't like ifdefs with exact systems. They should use more generic ifdefs, the ones that actually say about the feature to be present. E.g. if this pthread API is available on most UNIXes then it should use negative logic in order to work in a maximum portable way.